### PR TITLE
South Africa Class 1 - Khayelihle Buthelezi - HTML/CSS - Week 2

### PR DESCRIPTION
**Pull request by Khaya56**
# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

### CODE REVIEW

**Comment by Khaya56** [comment]
Still need to make it responsive using media queries, will ask about this in class

File: css/style.css [hunk 1]
```diff
@@ -4,7 +4,114 @@
...
+.container{
+    clear: both;
+    overflow: auto;
+}
+
+nav{float:right;}
+
+.logo img{
+    position: relative;
+    float: left;
```
Comment (LucyMac) [inline css/style.css:23]: Avoid using `float` and `clear` rules. This is antiquated and has been replaced by flexbox. Flexbox offers a more reliable way of laying out elements, and once you get used to it it'll become very easy 🙂

Comment (LucyMac) [reply css/style.css:23]: Remove the float rules and change your `container` rules to use flexbox. Do you remember how to do this? Have a look [here](https://www.freecodecamp.org/news/an-animated-guide-to-flexbox-d280cf6afc35/) if not 😉

File: css/style.css [hunk 2]
```diff
@@ -4,7 +4,114 @@
...
+   padding:10px;
+}
+
+ul li:hover{
+  color:orange;
+  
+}
+
+.bckimg{
+    background-image: url(/img/first-background.jpg);
```
Comment (LucyMac) [inline css/style.css:39]: Did this path work for you? It doesn't for me. I've had to add 2 dots for it to work: `../img/first-background.jpg`

File: css/style.css [hunk 3]
```diff
@@ -4,7 +4,114 @@
...
+    font-size: 30px;
+    color: white;
+    text-align: center;
+}
+.btn {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    -ms-transform: translate(-50%, -50%);
```
Comment (LucyMac) [inline css/style.css:57]: Nice trick but you shouldn't need this. Change the container (`bckimg`) to use flexbox and center everything inside it to achieve the same result without using `position: absolute;`

File: css/style.css [hunk 4]
```diff
@@ -4,7 +4,114 @@
...
+    cursor: pointer;
+    border-radius: 5px;
+    text-align: center;
+  }
+  .services p{
+      font-size: 30px;
+      text-align: center;
+  }
+  .column {
+    float: left;
```
Comment (LucyMac) [inline css/style.css:72]: Again, can you position these without using floats?

File: css/style.css [hunk 5]
```diff
@@ -4,7 +4,114 @@
...
+    text-align: center;
+  }
+
+.img{
+    display: block;
+    width: 30%;
+    margin-left: auto;
+    margin-right: auto;
+}
+.Footer{
```
Comment (LucyMac) [inline css/style.css:83]: we don't usually use upper case letters in class names.

File: index.html [hunk 6]
```diff
@@ -12,6 +12,66 @@
...
 </head>
 <body>
+    <!--Navbar-->
+    <div class="container">
+        <div class="logo">
+          <img src="/img/karma-logo.svg" alt="logo">
+          </div>
+        <nav>
+          <ul>
+          <li><b>Meet Karma</b></li>
```
Comment (LucyMac) [inline index.html:22]: Good semantics.
Watch your indentation. Good formatting makes codes much easier to read, and errors much easier to identify.

File: index.html [hunk 7]
```diff
@@ -12,6 +12,66 @@
...
+          <li>Login</li>
+          </ul>
+        </nav>
+    </div>
+
+        <div class="bckimg">
+            <br>
+            <br>
+            <br>
+            <br>
```
Comment (LucyMac) [inline index.html:36]: Don't rely on HTML to create space on your page. This is the job of CSS. What type of spacing can you create to add room inside `bckimg`?

File: index.html [hunk 8]
```diff
@@ -12,6 +12,66 @@
...
+            <p>Introducing Karma</p>
+            <p>Bring WiFi with you,everywhere you go.</p>
+            <button class="btn">Learn More</button>
+        </div>
+        <br>
+        <br>
+        <!--Services-->
+        <div class="services">
+            <p>Everyone needs a little Karma.</p>
```
Comment (LucyMac) [inline index.html:45]: Is this body content or a title? If it's a title, it should use one of the header tags `h1`, `h2`, `h3` etc.

File: index.html [hunk 9]
```diff
@@ -12,6 +12,66 @@
...
+                  <p>Internet for all devices</p>
+            </div>
+          
+            <div class="column">
+                <img class="img" src="/img/icon-coffee.svg" alt="Mike">
+                  <p>Boost your productivity</p>
+            </div>
+            
+            <div class="column">
+                <img class="img" src="/img/icon-refill.svg" alt="John">
```
Comment (LucyMac) [inline index.html:59]: Who are Jane, Mike and John? 😄  Remember to use descriptive text inside the alt attribute.

**Review by LucyMac — COMMENTED** [review]
Nice work however please replace all uses of `float` with flexbox alternatives. It enables much more reliable and simple layouts and makes changing layouts in different screen sizes much easier.

**Comment by LucyMac** [comment]
> Still need to make it responsive using media queries, will ask about this in class

This is the topic of tomorrow's class so don't worry 🙂

**Comment by LucyMac** [comment]
I'm sorry I accidentally reviewed your PR thinking you were one of my students - I hope the comments are useful anyway! 😄

**Comment by Khaya56** [comment]
> I'm sorry I accidentally reviewed your PR thinking you were one of my students - I hope the comments are useful anyway! 😄

Very much so, thank you very much

**Pull request closed without merging**